File: Lecture 6/FinalCode - Java/app/src/main/java/com/inovaeg/androidtrainingpharos2020/RequestInterface.java
package com.inovaeg.androidtrainingpharos2020;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;
import retrofit2.http.Query;

// TODO (6.3) define requests interface
public interface RequestInterface {

    @GET("authentication/guest_session/new")
    Call<GuestSession> createGuestSession(@Query("api_key") String apiKey);

    @GET("movie/top_rated")
    Call<Movies> getTopRatedMovies(@Query("api_key") String apiKey, @Query("page") int pageNumber);

    @POST("movie/{movie_id}/rating")
    Call<RatingResponse> rateMovie(@Path("movie_id") int movieId, @Query("api_key") String apiKey, @Body RatingRequest request, @Query("guest_session_id") String sessionId);
}

File: Lecture 6/FinalCode - Kotlin/app/src/main/java/com/inovaeg/androidtrainingpharos2020/RequestInterface.kt
package com.inovaeg.androidtrainingpharos2020

import retrofit2.Call
import retrofit2.http.*

// TODO (6.3) define requests interface
interface RequestInterface {
    @GET("authentication/guest_session/new")
    fun createGuestSession(@Query("api_key") apiKey: String?): Call<GuestSession?>?

    @GET("movie/top_rated")
    fun getTopRatedMovies(@Query("api_key") apiKey: String?, @Query("page") pageNumber: Int): Call<Movies?>?

    @POST("movie/{movie_id}/rating")
    fun rateMovie(@Path("movie_id") movieId: Int, @Query("api_key") apiKey: String?, @Body request: RatingRequest?, @Query("guest_session_id") sessionId: String?): Call<RatingResponse?>?
}